src/plugins/data/public/search/session/search_session_state.ts

```diff
@@ -104,6 +104,11 @@ export interface SessionStateInternal<SearchDescriptor = unknown> {
    * Start time of current session
    */
   startTime?: Date;
+
+  /**
+   * Should responses be cached on the client
+   */
+  cacheOnClient?: boolean;
```

Comment:
I now wonder, why we make this option a session-related thing and not just an option on a search request? 

`data.search(request, {cacheOnClient: true})`

Reply:
Because caching is always a session attribute.
The session is being cached, not each individual search request.
Otherwise, you'd have to make sure that this flag is sent in together with a sessionId

Reply:
I don't understand 🤔 : 

> The session is being cached, not each individual search request


In this pr on a client: 
Aren't we caching individual search requests? As I understand we are.
Can't we do the same thing without `sessionId`?  I think we can. 


> Otherwise, you'd have to make sure that this flag is sent in together with a sessionId

If the only reason why we are tieing this client-side caching to a `sessionId` is to simplify search options passing, then I think we shouldn't. 🤔 
IMO we should allow using this caching without `sessionId` and not rely on it internally as these seem to be independent not intersecting mechanics. 

Maybe I am missing some context here

Reply:
i taught we are coupling it to sessions as that would make it clear when the cache should be removed (on creating a new session) but i see the current implementation uses a timeout, so wondering as well, what are the benefits of coupling this to sessions?

Reply:
It's not going to use a timeout (See joe's comment. It was an intermediate step)